### ISSUE TITLE
Special Characters in email or name/alias cause login to fail

### ISSUE THREAD

**Issue by Starker3** [piwik/plugin-LoginLdap#293]
When users are synchronised using this plugin, it seems to convert special characters in both the email address as well as the alias, for example:
![image](https://user-images.githubusercontent.com/20905821/187802081-66833352-6534-4e9d-8cc0-d6136262cae2.png)
(In this case the surname is O'Brien which was used to generate the alias)

This then results in that user not being able to login to Matomo with a "Wrong password or email" error even when the user name is valid (I.e. just alphanumeric characters):
![image](https://user-images.githubusercontent.com/20905821/187802158-51b6001b-a526-4c23-8f9e-4217e620ea1f.png)

But since the alias field is not directly editable in the UI this has to be fixed by updating the record in the database directly using a SQL query. Only the email can be edited.
![image](https://user-images.githubusercontent.com/20905821/187802364-f769c4fb-b57f-4bc4-84e5-71165d0e406c.png)

Partially related to: https://github.com/matomo-org/matomo/issues/11796
But since the LoginLdap plugin directly inserts the data into the database, it is possible to create an account using a special character but it is then impossible to login as that user.

**Comment by AltamashShaikh** [comment]
@Starker3 I was able to run and figure this out, after authenticating via Ldap plugin we try to update the current user info without password and for that we make a request to `UsersManager.updateUser` and sent the email has for eg `johndoe's@domain.com` 
But the value received by core is `johndoe&#039;s@domain.com` and it tries to update the email since the one we have stored in DB is `johndoe's@domain.com` and its different, but `johndoe&#039;s@domain.com`  is not a valid email and it throws an exception `email: The value "johndoe&#039;s@domain.com" does not look like a valid email.` but the final message we show in  UI is `Error: Wrong username and/or password.`


@sgiehl Its due to this [line](https://github.com/matomo-org/matomo/blob/4.x-dev/plugins/UsersManager/API.php#L923) we are detecting the change in `UsersManager` plugin.

If I update it to
```
$hasEmailChanged = mb_strtolower($email) !== mb_strtolower($userInfo['email']) && mb_strtolower(html_entity_decode($email,ENT_COMPAT|ENT_QUOTES, 'UTF-8')) !== mb_strtolower($userInfo['email']);
```
We are able to login

**Comment by sgiehl** [comment]
I guess that's the "old" problem with having all input parameters sanitized by default. 
@AltamashShaikh Adding a line like `$email = Common::unsanitizeInputValue($email);` should also solve that.
Feel free to create a PR for that.

**Comment by AltamashShaikh** [comment]
Created - https://github.com/matomo-org/plugin-LoginLdap/issues/293

**Comment by AltamashShaikh** [comment]
@Starker3 should be fixed with release of Matomo 4.12
